### ISSUE TITLE
quick reference for different types of tasks

### ISSUE THREAD

**Issue by edublancas** [ploomber/ploomber#631]
from @fferegrino:

> I think having a section where the syntax/structure for each task is displayed is a good idea, for example, shell tasks have their own top level section: https://docs.ploomber.io/en/latest/user-guide/shell.html whereas Python tasks do not: https://docs.ploomber.io/en/latest/get-started/basic-concepts.html#tasks-scripts-notebooks

I think this is a great idea. But I'm unsure how to place it in the docs. Perhaps add a "Task types" document in the "Cookbook" section showing a minimal example for each task type:

* script/notebook
* function
* SQL script
* shell script

Then have each relevant section in the user guide (which is more explanatory in nature) link to the cookbook for people who want to copy-paste something.

e.g. the user guide that explains how to use the shell script should link at the top to the cookbook that has a quick example on using shell tasks

thoughts? @fferegrino

**Comment by idomic** [comment]
What about separating the two? Usually docs, for instance [Sagemakers](https://docs.aws.amazon.com/sagemaker/latest/dg/api-and-sdk-reference.html) has this differentiation between new-commers and users who know what they're doing.

1. Should cover the get started.
2. Should be use case specific (with copy-paste examples).
3. A complete CLI/SDK reference guide (for deeper dives).

**Comment by edublancas** [comment]
Yep, [I like this](https://documentation.divio.com/) to guide our doc design

In our docs:

1. user guide -> learning
2. cookbook -> quick stuff, copy-paste
3. API reference -> details
